src/network/core.rs

```diff
@@ -0,0 +1,172 @@
+use std::fs::File;
+use std::os::unix::prelude::*;
+use ipnetwork::IpNetwork;
+use futures::stream::TryStreamExt;
+use rtnetlink;
+use rtnetlink::Error;
+
+pub struct Core {
+    pub networkns: String,
+}
+
+impl Core {
+    // adds specified gateway to interface
+    pub async fn add_gateway_address(
+        handle: &rtnetlink::Handle,
+        ifname: &str,
+        ip: &IpNetwork,
+    ) -> Result<(), Error> {
+        let mut links = handle
+            .link()
+            .get()
+            .set_name_filter(ifname.to_string())
+            .execute();
+        if let Some(link) = links.try_next().await? {
+            handle
+                .address()
+                .add(link.header.index, ip.ip(), ip.prefix())
+                .execute()
+                .await?
+        }
+        Ok(())
+    }
+
+    // turn up the interface
+    pub async fn set_link_up(handle: &rtnetlink::Handle, ifname: &str) -> Result<(), Error> {
```

Comment:
We can perform sync block on a higher level caller function. If we make low level calls `sync` early we will loose option to make code `async`. So i thought it would be a good optional thing to have.

Reply:
@mheon Convert the main calls to `block_on`.